### DISCUSSION TITLE
Embeddings

### DISCUSSION THREAD

**Discussion by holen82** [marcominerva/ChatGptNet#103] [Ideas]
Any plans for adding support for using OpenAI API for generating embeddings? OpenAIChatGptServiceConfiguration.cs is hardcoded to use the [/chat/completions](https://platform.openai.com/docs/api-reference/completions) endpoint no matter what model is passed in. Is it an idea the check what model is passed, and when for example the "text-embedding-ada-002" model is passed, it could return the [/embeddings](https://platform.openai.com/docs/api-reference/embeddings) endpoint of the ChatGPT API?

**Comment by marcominerva** [comment]
The **ChatGptNet** library is tied to chat completion, offering some specific features like response streaming, function calling and automatic conversation handling. Embeddings requires different behaviors and object models. Instead of modifying the existing methods, I think it could be a better idea to provide something that explicitly targets embeddings.

I will think about it!

**Reply by holen82** [reply]
Having looked further into it I absolutely agree. I've played some with the code, and think an CreateEmbeddingAsync method in ChatGtpClient makes sense. To create the request I've written a CreateEmbeddingsRequest method, and added a GetEmbeddingsEndpoint method in ChatGptServiceConfiguration. Also, for when reading the response one will need to implement a ChatGptEmbeddingResponse(and potentially create a NormalizeEmbeddingResponse method).

**Reply by marcominerva** [reply]
So, what about making a PR? :)

**Reply by holen82** [reply]
PR created 🙂
